core/node/l1_recovery/src/l1_fetcher/l1_fetcher.rs

```diff
@@ -0,0 +1,1083 @@
+use std::{cmp, cmp::PartialEq, collections::HashMap, future::Future, sync::Arc};
+
+use anyhow::{anyhow, Result};
```

Comment:
use qualified `anyhow::Result` instead, so as not to shadow the Result from prelude, especially since you define custom errors in this file as well.